python/katana/dynamic_bitset.pyx

```diff
@@ -0,0 +1,56 @@
+from .cpp.libgalois.DynamicBitset cimport DynamicBitset as CDynamicBitset
+
+
+cdef class DynamicBitset:
+
+    cdef CDynamicBitset underlying
+
+    def __cinit__(self):
+        self.underlying = CDynamicBitset()
+
+    def resize(self, n):
+        """Resizes the bitset."""
+        self.underlying.resize(n)
+
+    def reserve(self, n):
+        """Reserves capacity for the bitset."""
+        self.underlying.reserve(n)
+
+    def clear(self):
+        """Clears the bitset."""
+        self.underlying.clear()
+
+    def shrink_to_fit(self):
+        """Shrinks the allocation for bitset to its current size."""
+        self.underlying.shrink_to_fit()
+
+    def size(self):
+        """Gets the number of bits held by the bitset."""
+        return self.underlying.size()
+
+    def reset(self):
+        """Unset every bit in the bitset."""
+        self.underlying.reset()
+
+    def reset_begin_end(self, begin, end):
+        """Unset a range of bits given an inclusive range."""
+        self.underlying.reset(begin, end)
```

Comment:
You should be able to roll this into `__setitem__` since that can take a slice which has a begin and end. You can fail in cases where the stride is not 1.

Reply:
done